test/mysql.discover.test.js

```diff
@@ -199,6 +199,23 @@ describe('Discover model primary keys', function() {
   });
 });
 
+describe('Discover user model with index', function() {
```

Comment:
can you please add some more test cases to cover more negative scenarios and other worst cases ?